drivers/sub8_videoray_m5_thruster/nodes/thruster_driver.py

```diff
@@ -238,10 +238,27 @@ def thrust_cb(self, msg):
     def alert_thruster_unloss(self, thruster_name):
         if thruster_name in self.failed_thrusters:
             self.failed_thrusters.remove(thruster_name)
-        
         if len(self.failed_thrusters) == 0:
             self.thruster_out_alarm.clear_alarm(parameters={"clear_all"})
         else:
+            rospy.wait_for_service('update_thruster_layout')
+            try:
+                update = rospy.ServiceProxy('update_thruster_layout', UpdateThrusterLayout)
+                rospy.wait_for_service('b_matrix')
+                try:
+                    B = rospy.ServiceProxy('b_matrix', BMatrix)
+                    B = np.asmatrix(B)
+                    if (np.linalg.matrix_rank(B) < 6):
+                        self.thruster_out_alarm.raise_alarm(
+                            parameters={
+                                'thruster_names': self.failed_thrusters + "B MATRIX IS SINGULAR",
+                            },
+                            severity=5
+                        )
+                except rospy.ServiceException, e:
+                    print "Service call failed: %s"%e
```

Comment:
This should be a rospy.logwarn/rospy.logerr so it shows up in logs and should describe what service call is failing